### DISCUSSION TITLE
what is the safest way to remove good_job?

### DISCUSSION THREAD

**Discussion by laptopmutia** [bensheldon/good_job#1535] [Q&A] [answered]
is removing the line from gemfile and removing the database tables enough?

my current project deployed using capistrano and passenger as in this tutorial https://gorails.com/deploy/ubuntu/24.04

but I want to migrate to kamal for simplicty, and currently don't know how to deploy goodjob with kamal 

So I plan to remove it from my project and use rails/solid_queue instead for more seamless experience

no malintent for the projects I really appreciate and grateful for it.
 since its been there to removing the cost and complexity for my apps.

**Comment by bensheldon** [comment, marked as answer]
No worries. I don’t think GoodJob requires anything special for deployment differently than Solid Queue (run the good_job executable), but I understand if you just want to follow what’s closest to Rails.

The safest way would be to leave GoodJob in place in the Gemfile and Database, and instead install Solid Queue and change your `config.queue_adapter = :solid_queue` to get all of the jobs running while _both_ are in place. You’ll need to doublecheck you aren’t using any GoodJob extensions (like Concurrency) or special behavior (like Batches), and switch over any Cron-like configuration.

Only once you’re sure no jobs are being enqueued or executed on GoodJob should you then:

- Remove from Gemfile
- Drop database tables
- Search your codebase for any mention of `good_job` or `GoodJob` and remove that. I’ve tried to keep all of the extensions and configuration keep the name so it’s easy to find what’s custom to GoodJob.
